### PR TITLE
BUG: Lilliefors min nobs not set

### LINKED ISSUE #5333: Issue with lilliefors after upgrading to 0.9.0 from 0.8.0 (regression?)
I had some code running with v0.8.0, but it won't work anymore with version 0.9.0. The issue can be reproduced simply with the following data and code:

```
from statsmodels.stats.diagnostic import lilliefors
lilliefors([18.50117949, 18.24272814, 18.50674601])
```

With 0.8.0 I just got a pandas warning when importing
```
/usr/lib64/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.                     
  from pandas.core import datetools
```

But now, I get an error when actually running lilliefors on those data:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/testone/lib64/python3.6/site-packages/statsmodels/stats/_lilliefors.py", line 344, in kstest_fit
    pval = lilliefors_table.prob(d_ks, nobs)
  File "/home/testone/lib64/python3.6/site-packages/statsmodels/stats/tabledist.py", line 120, in prob
    critv = self._critvals(n)
  File "/home/testone/lib64/python3.6/site-packages/statsmodels/stats/tabledist.py", line 99, in _critvals
    return np.array([p(n) for p in self.polyn])
  File "/home/testone/lib64/python3.6/site-packages/statsmodels/stats/tabledist.py", line 99, in <listcomp>
    return np.array([p(n) for p in self.polyn])
  File "/usr/lib64/python3.6/site-packages/scipy/interpolate/polyint.py", line 79, in __call__
    y = self._evaluate(x)
  File "/usr/lib64/python3.6/site-packages/scipy/interpolate/interpolate.py", line 634, in _evaluate
    below_bounds, above_bounds = self._check_bounds(x_new)
  File "/usr/lib64/python3.6/site-packages/scipy/interpolate/interpolate.py", line 663, in _check_bounds
    raise ValueError("A value in x_new is below the interpolation "
ValueError: A value in x_new is below the interpolation range.
```

I checked the doc, but I did not see any specific instruction on how data should be: is this a regression or am I actually providing wrong data? In the latter case, can you refer me to the relative doc?

Thanks

### PR DESCRIPTION
**Pull request by bashtage**
- [x] closes #5333
- [X] tests added / passed. 
- [x] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message).

### CODE REVIEW

**Comment by bashtage** [comment]
@josef-pkt You should take a look at this fix when you get a chance.

File: statsmodels/stats/tests/test_lilliefors.py [hunk 1]
```diff
@@ -57,3 +57,15 @@ def test_pval_bounds(self):
...
+        x = np.arange(3.)
+        with pytest.raises(ValueError):
+            lilliefors(x, dist='norm')
+        x = np.arange(2.)
+        with pytest.raises(ValueError):
+            lilliefors(x, dist='exp')
+
+    def test_large_sample(self, reset_randomstate):
+        x = np.random.randn(1500)
+        lilliefors(x)
```
Comment (josef-pkt) [inline statsmodels/stats/tests/test_lilliefors.py]: this needs at least a regression test for the values returned by lilliefors

**Comment by josef-pkt** [comment]
I'm not sure yet about the fix
The PR is hard to look at. Because of the style changes, it is not obvious what has been changed in substance.

AFAIU, we need to enable users to get the option of using either method for critical/p-values in #5333
(e.g. Can we compare Stephen's approximation and the table approximation for large samples?)
I haven't looked enough to understand the details again.

**Comment by josef-pkt** [comment]
Fix for large nobs here looks to be roughly as discussed in #5333, but maybe we can skip interpolation if it is just a function of nobs for values above table. (but I guess the very large nobs range won't matter much)

AFAICS, Stephen's approximation is not used in this function, maybe that was for Anderson-Darling.

**Pull request merged**